design/src/docs/anatomy.md

```diff
@@ -0,0 +1,687 @@
+            Ok(ok) => ok.oneshot(req).await,
+            // Failed to route, use the `R::Error`s `IntoResponse<P>`.
+            Err(error) => {
+                debug!(%error, "failed to route");
+                Err(Box::new(error.into_response()))
+            }
+        }
+    }
+}
+```
+
+The `RouterService` is the final piece necessary to form a functioning composition - it is used to aggregate together the HTTP services, created via the upgrade procedure, into a single HTTP service which can be presented to the customer.
+
+![RouterService](imgs/router.png)
+
+## Builders
+
+The service builder is the primary public API, it can be generated for every [Smithy Service](https://awslabs.github.io/smithy/2.0/spec/service-types.html). At a high-level, the purpose of a service builder is to take a function for each Smithy Operation, whose signature is constrained by the Smithy model, and produce a single HTTP service.
+
+```rust
+/// The service builder for [`PokemonService`].
+///
+/// Constructed via [`PokemonService::builder`].
+pub struct PokemonServiceBuilder<Op1, Op2, Op3, Op4, Op5, Op6> {
+    capture_pokemon_operation: Op1,
+    empty_operation: Op2,
+    get_pokemon_species: Op3,
+    get_server_statistics: Op4,
+    get_storage: Op5,
+    health_check_operation: Op6,
+}
+```
+
+The builder has one generic variable for each [Smithy Operation](https://awslabs.github.io/smithy/2.0/spec/service-types.html#operation) in the [Smithy Service](https://awslabs.github.io/smithy/2.0/spec/service-types.html#service). Each setter switches the type of the `Op{N}` type parameter:
+
+```rust
+    /// Sets the [`GetPokemonSpecies`](crate::operation_shape::GetPokemonSpecies) operation.
+    ///
+    /// This should be an [`Operation`](aws_smithy_http_server::operation::Operation) created from
+    /// [`GetPokemonSpecies`](crate::operation_shape::GetPokemonSpecies) using either
+    /// [`OperationShape::from_handler`](aws_smithy_http_server::operation::OperationShapeExt::from_handler) or
+    /// [`OperationShape::from_service`](aws_smithy_http_server::operation::OperationShapeExt::from_service).
+    pub fn get_pokemon_species_operation<NewOp>(
+        self,
+        value: NewOp,
+    ) -> PokemonServiceBuilder<
+        Op1,
+        Op2,
+        NewOp,
+        Op4,
+        Op5,
+        Op6,
+    > {
+        PokemonServiceBuilder {
+            capture_pokemon_operation: self.capture_pokemon_operation,
+            empty_operation: self.empty_operation,
+            get_pokemon_species: value,
+            get_server_statistics: self.get_server_statistics,
+            get_storage: self.get_storage,
+            health_check_operation: self.health_check_operation,
+        }
+    }
+
+    /// Sets the [`GetPokemonSpecies`](crate::operation_shape::GetPokemonSpecies) operation.
+    ///
+    /// This should be a closure satisfying the [`Handler`](aws_smithy_http_server::operation::Handler) trait.
```

Comment:
Should we change these docs to highlight this can also be a function? And to mention `async`.